llvm/lib/Target/AIE/AIEMaxLatencyFinder.cpp

```diff
@@ -131,6 +129,46 @@ int MaxLatencyFinder::findEarliestRef(
   return Cycle;
 }
 
+/// Whether the SUnits in the DAG have know successor instructions. Those
+/// can be from successor MBBs, or from a fall-through timed region inserted
+/// by the AIEPostPipeliner.
+bool hasScheduledSuccessorInstructions(MachineBasicBlock::iterator RegionEnd,
+                                       MachineBasicBlock &MBB,
+                                       const AIEPostRASchedStrategy &Sched) {
+  const BlockState &BS = Sched.getInterBlock().getBlockState(&MBB);
+  return isBottomRegion(RegionEnd, MBB) &&
+         (!BS.BottomInsert.empty() || Sched.successorsAreScheduled(&MBB));
+}
+
+/// Return the list of scheduled instructions that will follow those in the DAG.
```

Comment:
Maybe DAG is out of context here (is a higher level artifact from the calling part).